solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java

```diff
@@ -531,7 +533,20 @@ private  SimpleOrderedMap getRangeCountsIndexed() throws IOException {
   private Query[] filters;
   private DocSet[] intersections;
   private void rangeStats(Range range, int slot) throws IOException {
-    Query rangeQ = sf.getType().getRangeQuery(null, sf, range.low == null ? null : calc.formatValue(range.low), range.high==null ? null : calc.formatValue(range.high), range.includeLower, range.includeUpper);
+    final Query rangeQ;
+    {
+      final Query rangeQuery = sf.getType().getRangeQuery(null, sf, range.low == null ? null : calc.formatValue(range.low), range.high==null ? null : calc.formatValue(range.high), range.includeLower, range.includeUpper);
+      if (fcontext.cache) {
+        rangeQ = rangeQuery;
+      } else if (rangeQuery instanceof ExtendedQuery) {
+        ((ExtendedQuery) rangeQuery).setCache(fcontext.cache);
```

Comment:
Here (and in the else) I think I would explicitly do `setCache(false)` as it feels more readable to me, but I don't have strong opinions on that.